contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,933 @@
+                     *
+                     * Also sqrtPriceX96 is 0 because the pool is already created
+                     * non zero amount attempts to create a new instance of the pool
+                     */
+                    INonfungiblePositionManager.MintParams({
+                        token0: WETH,
+                        token1: OETHb,
+                        tickSpacing: tickSpacing,
+                        tickLower: lowerTick,
+                        tickUpper: upperTick,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        recipient: address(this),
+                        deadline: block.timestamp,
+                        sqrtPriceX96: 0
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+            tokenId = _mintedTokenId;
+        } else {
+            (
+                ,
+                uint256 __wethAmountSupplied,
+                uint256 __oethbAmountSupplied
+            ) = positionManager.increaseLiquidity(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     */
+                    INonfungiblePositionManager.IncreaseLiquidityParams({
+                        tokenId: tokenId,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+        }
+
+        _updateUnderlyingAssets();
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            _oethbRequired, // oethbAmountDesired
+            _wethAmountSupplied, // wethAmountSupplied
+            _oethbAmountSupplied, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the Aerodrome pool price is within the expected
+     *      parameters.
+     *      This function ignores whether the strategy contract has liquidity
+     *      position in the pool.
+     */
+    function _checkForExpectedPoolPrice() internal {
+        uint160 _currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+
+        if (_currentPrice < sqrtRatioX96Tick0 ||
```

Comment:
🔴 This allows invalid values past, since we should let the tick edges through.

Reply:
Agree with you. 
All the more so this could lead to the following call, `estimateAmount1()` to revert as the current price can be equal to the price of the edge of the tick. 
There is a div in `estimateAmount1()` with `currentPrice-priceAtTick(-1 or 0), and these two values can be equal, (see image).

![Capture d’écran 2024-08-22 à 23 33 16](https://github.com/user-attachments/assets/69523591-83fa-4936-a81b-fb771b676ac3)

Finally, as the expect price should always be between ticks  -1 and 0 expected, we should maybe use 
```
if (_currentPrice < sqrtRatioX96Tick0 ||  _currentPrice > sqrtRatioX96Tick1)
```

Reply:
Yes this has been changed in one of the previous commits to also revert on ticker edges.